user-docs/documentation/docs/getting-started/5-add-a-new-serverless-contract-library.mdx

```diff
@@ -0,0 +1,205 @@
+---
+sidebar_position: 5
+---
+
+# Add a new serverless contract
+
+Serverless contracts are a central part of Swarmion. They let you define interactions between your services to enable more stability and type safety. You can learn more in the [dedicated documentation section](../how-to-guides/use-serverless-contracts)
+
+In this section, we will create a contract library for the users backend, and how to use it.
+
+## Prerequisites
+
+For this section, we will suppose that there is an existing frontend service in the monorepo application:
+
+import ThemedImage from '@theme/ThemedImage';
+import swarmionAppFrontendGraphLight from '../../static/screenshots/swarmion-app-frontend-graph-light.png';
+import swarmionAppFrontendGraphDark from '../../static/screenshots/swarmion-app-frontend-graph-dark.png';
+
+<div
+  style={{
+    width: '100%',
+    display: 'flex',
+    'justify-content': 'center',
+    'margin-bottom': 'var(--ifm-leading)',
+  }}
+>
+  <ThemedImage
+    alt="Swarmion app with frontend"
+    sources={{
+      light: swarmionAppFrontendGraphLight,
+      dark: swarmionAppFrontendGraphDark,
+    }}
+  />
+</div>
+
+:::info
+The `generate-frontend-service` command is a planned development by the Swarmion core team.
+:::
+
+## Generate and link the contract library
+
+First, lets generate the library, with the generator we used in the previous section:
+
+``` bash
+yarn generate-library users-contracts --directory contracts
+```
+
+:::info
+The `--directory` option let you chose the path containing your generated library. Default value is `packages`
+:::
+
+Then, we will add this contract library as a dependency for the users-backend and the frontend:
+
+```bash
+cd backend/users
+yarn add @my-project-name/users-contracts
+
+cd ../../frontend
+yarn add @my-project-name/users-contracts
+```
+
+The project dependency graph should now looks like the following:
+
+import swarmionAppUsersContractsGraphLight from '../../static/screenshots/swarmion-app-users-contracts-graph-light.png';
+import swarmionAppUsersContractsGraphDark from '../../static/screenshots/swarmion-app-users-contracts-graph-dark.png';
+
+<div
+  style={{
+    width: '100%',
+    display: 'flex',
+    'justify-content': 'center',
+    'margin-bottom': 'var(--ifm-leading)',
+  }}
+>
+  <ThemedImage
+    alt="Swarmion app with users contracts lib"
+    sources={{
+      light: swarmionAppUsersContractsGraphLight,
+      dark: swarmionAppUsersContractsGraphDark,
+    }}
+  />
+</div>
+
+## Define our first contract
+
+Now lets write our first contract to define a get user operation.
```

Comment:
We should add a note/section to explain that the contract must be exported by the `src/index.ts` file